### PR TITLE
Small Fixes

### PR DESCRIPTION
**Pull request by h-arslan**
Thanks for the excellent work you've done. 
I have added small fixes on training loop in using_bmdal notebook and downloading data.

### CODE REVIEW

**Comment by dholzmueller** [comment]
Thank you for the fixes. 
The OpenML fix doesn't seem to be doing what it should do, since newer OpenML versions have renamed the variable and the corresponding function. I pushed another fix instead which calls the renamed function `set_root_cache_directory` if available.
The fix for the training loop is not necessary if I am not mistaken, both versions should work, assuming the gradients are zero before the training.
For the reasons above it is not necessary anymore to merge this pull request.
